src/pybind/mgr/dashboard/services/rbd.py

```diff
@@ -360,6 +360,10 @@ def _rbd_image(cls, ioctx, pool_name, namespace, image_name,  # pylint: disable=
             # snapshots
             stat['snapshots'] = []
             for snap in img.list_snaps():
+                # Skip trash snapshots (cloned-and-then-deleted format v2 snapshots)
+                if snap['namespace'] == rbd.RBD_SNAP_NAMESPACE_TYPE_TRASH:
```

Comment:
Yesterday I said that
> Hiding these snapshots such that there is no trace of them at all is probably not a good idea

but, going by the screenshot, this appears to do exactly that...  How about:

- skipping `img.is_protected_snap()` call in this case, just like it's skipped in case of snapshot-based mirroring
- replacing `img.set_snap(snap['name'])` call with `img.set_snap_by_id(snap['id'])`

I'm wondering if `img.list_children()` would "just work" in that case -- displaying such a snapshot and its children is much better than hiding all of that.

Reply:
> * skipping `img.is_protected_snap()` call in this case, just like it's skipped in case of snapshot-based mirroring

If I'm reading this right, `snap['is_protected']` is left to be `None` in case snapshot-based mirroring.  This could be improved -- since only regular user snapshots can be protected, I'd write it as follows:
```
snap['is_protected'] = False
if snap['namespace'] == rbd.RBD_SNAP_NAMESPACE_TYPE_USER:
    snap['is_protected'] = img.is_protected_snap(snap['name'])
```
This would cover all types of snapshots in a generic way.

Reply:
> Hiding these snapshots such that there is no trace of them at all is probably not a good idea

Yup, @idryomov, I knew, but given this should be backported ASAP to all active releases I wanted:
* A minimal/portable change (no changes to the API),
* Something that matches the default output of `rbd snap ls`.

If I don't do this way, whenever a user deletes a snapshot with clones, they'll continue seeing that snapshot. I'd rather display the deleted snapshot in the Trash section:

![image](https://github.com/ceph/ceph/assets/37327689/8b39aaa1-db6e-456d-ad4c-49044f874054)

But `rbd trash` only works for images, right?

BTW, (CC @nizamial09) IMHO the `/api/block/{image}` endpoint tries to do too many things (iterate through all snapshots and children, calculate the real Image usage with `diff_iterate()`), which results in a very heavyweight endpoint. We should probably add endpoints for the snapshots (`/api/block/{image}/snapshot`) and a new query param (`minimal=<bool>`) to retrieve only the basic RBD Image info.

Reply:
BTW @idryomov I'll apply this snippet you shared. Thanks!:

```python
snap['is_protected'] = False
if snap['namespace'] == rbd.RBD_SNAP_NAMESPACE_TYPE_USER:
    snap['is_protected'] = img.is_protected_snap(snap['name'])
```